src/server/engine_factory.rs

```diff
@@ -171,8 +173,35 @@ impl<ER: RaftEngine> KvEngineFactory<ER> {
 
 impl<ER: RaftEngine> TabletFactory<RocksEngine> for KvEngineFactory<ER> {
     #[inline]
-    fn create_tablet(&self) -> Result<RocksEngine> {
+    fn create_root_db(&self) -> Result<RocksEngine> {
         let root_path = self.kv_engine_path();
-        self.create_tablet(&root_path)
+        let tablet = self.create_tablet(&root_path)?;
+        let mut root_db = self.inner.root_db.lock().unwrap();
+        root_db.replace(tablet.clone());
+        Ok(tablet)
+    }
+
+    fn create_tablet(&self, _id: u64, _suffix: u64) -> Result<RocksEngine> {
```

Comment:
All the following methods are useless for v1 factory as it should not be used at all.

Reply:
Once we finished the raftstore2, let's revisit them if they're truly useless, I will separate TabletFactory into two, one is for v1 and the other is for v2. 
Right now let's unify them as it could be used potentially to share the raftstore code.